tests/interpreters/test_python.py

```diff
@@ -169,3 +169,69 @@ def  score(input):
     return (np.asarray([1, 2])) * (1)
 """
     utils.assert_code_equal(interpreter.interpret(expr), expected_code)
+
+
+class CustomPythonInterpreter(interpreters.PythonInterpreter):
+    bin_depth_threshold = 2
+
+
+def test_depth_threshold_with_bin_expr():
+    expr = ast.NumVal(1)
+    for i in range(4):
+        expr = ast.BinNumExpr(ast.NumVal(1), expr, ast.BinNumOpType.ADD)
+
+    interpreter = CustomPythonInterpreter()
+
+    expected_code = """
+def  score(input):
+    var0 = (1) + ((1) + (1))
```

Comment:
Can we add a test case where `BinExpr` are diluted with some other exprs?
Like `BinNumExpr` -> `BinNumExpr` -> `SubroutineExpr`-> `BinNumExpr` -> `IfExpr` -> `BinNumExpr`. Something along those lines.